app/screens/thread/thread.tsx

```diff
@@ -0,0 +1,152 @@
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type PostModel from '@typings/database/models/servers/post';
+
+type ThreadProps = {
+    channel: ChannelModel;
+    componentId: string;
+    rootPost: PostModel;
+};
+
+const edges: Edge[] = ['left', 'right'];
+
+const getStyleSheet = StyleSheet.create(() => ({
+    flex: {
+        flex: 1,
+    },
+}));
+
+const CLOSE_BUTTON_ID = 'close-threads';
+
+const Thread = ({channel, componentId, rootPost}: ThreadProps) => {
+    const {formatMessage} = useIntl();
+    const appState = useAppState();
+    const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
+    const styles = getStyleSheet();
+    const theme = useTheme();
+
+    // Get post thread
+    useEffect(() => {
+        fetchPostThread(serverUrl, rootPost.id);
+    }, []);
+
+    const close = useCallback(() => {
+        dismissModal({componentId});
+        return true;
+    }, []);
+
+    const leftButton = useMemo(() => {
+        return {
+            id: CLOSE_BUTTON_ID,
+            icon: CompassIcon.getImageSourceSync('close', 24, theme.centerChannelColor),
+            testID: CLOSE_BUTTON_ID,
+        };
+    }, [isTablet, theme.centerChannelColor]);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case CLOSE_BUTTON_ID:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        let title;
+        if (channel?.type === General.DM_CHANNEL) {
+            title = formatMessage({id: 'thread.header.thread_dm', defaultMessage: 'Direct Message Thread'});
+        } else {
+            title = formatMessage({id: 'thread.header.thread', defaultMessage: 'Thread'});
+        }
+
+        let subtitle = '';
+        if (channel?.type !== General.DM_CHANNEL) {
+            subtitle = formatMessage({id: 'thread.header.thread_in', defaultMessage: 'in {channelName}'}, {channelName: channel?.displayName});
+        }
+
+        mergeNavigationOptions(componentId, {
+            topBar: {
+                leftButtons: [leftButton!],
+                title: {
+                    fontFamily: 'OpenSans-SemiBold',
+                    fontSize: 18,
+                    text: title,
+                },
+                subtitle: {
+                    color: changeOpacity(theme.sidebarHeaderTextColor, 0.72),
+                    text: subtitle,
+                },
+            },
+        });
```

Comment:
same with this, looks like we can assign this earlier. Same function to open the thread modal.

Reply:
Moving it to the local action is having issues with getting i18n & theme out of the component.

Reply:
@anurag6713 we talked about i18n in the meeting and for the theme you can use `EphemeralStore.theme` from `@stores/ephemeral_store`

Reply:
Yes, I remember the 18n .. Was listing the dependencies along with `Theme`. Sure, I will use theme from there .. but from code point of view, I think `mergeNavigationOptions` would still be here to handle `close` button and to handle `follow` button when CRT is implemented.

Reply:
So, I will be having `mergeNavigationOptions` in two places.

In `switchThread` to handle - Title, Subtitle
In `Thread` Screen to handle - Close Button & Follow button.

Not completely sure but I was thinking if it is affecting the readability or complicating things.

Reply:
I don't think it does.. one is to open the screen, the other is to update it.

in fact you should not be calling `mergeNavigationOptions` in the action, but pass the options to the `showModal` function

Reply:
Understood 👍 making the change. Thank you 🙂